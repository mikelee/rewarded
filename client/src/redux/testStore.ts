import { configureStore } from '@reduxjs/toolkit';
import rootReducer from './root-reducer';
import { FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER } from 'redux-persist';
import { ReduxState } from './store';

const testStoreState: ReduxState = {
    user: {
        currentUser: {
            userId: 1,
            username: 'Test User'
        },
        settings: {
            colorTheme: 'purple'
        }
    },
    todos: {
        todos: [
            {
                todoId: 1,
                text: 'Todo 1',
                completed: false,
                timestamp: '2023-03-08T06:49:33.913Z'
            },
            {
                todoId: 2,
                text: 'Todo 2',
                completed: true,
                timestamp: '2023-03-08T06:49:43.318Z'
            },
            {
                todoId: 3,
                text: 'Todo 3',
                completed: false,
                timestamp: '2023-03-08T06:49:47.183Z'
            }
        ]
    },
    rewards: {
        rewards: [
            {
                rewardId: 1,
                text: 'Reward 1',
                timestamp: '2023-03-08T06:49:53.064Z',
                completed: false
            },
            {
                rewardId: 2,
                text: 'Reward 2',
                timestamp: '2023-03-08T06:49:58.519Z',
                completed: true
            },
            {
                rewardId: 3,
                text: 'Reward 3',
                timestamp: '2023-03-08T06:50:03.270Z',
                completed: true
            }
        ],
        selectedRewardId: null
    },
    requirements: {
        requirements: [
            {
                rewardId: 1,
                todoId: 1,
                text: 'Todo 1',
                completed: false,
                timestamp: '2023-03-08T06:49:33.913Z'
            },
            {
                rewardId: 1,
                todoId: 2,
                text: 'Todo 2',
                completed: true,
                timestamp: '2023-03-08T06:49:43.318Z'
            },
            {
                rewardId: 2,
                todoId: 2,
                text: 'Todo 2',
                completed: true,
                timestamp: '2023-03-08T06:49:43.318Z'
            }
        ]
    },
    menu: {
        visible: false,
        sort: 'Newest First',
        submenuCategory: null
    },
    temporary: {
        loggedOutMessage: false
    },
    _persist: {
        version: -1,
        rehydrated: false
    }
}

export const testStore = configureStore({
    reducer: rootReducer,
    preloadedState: testStoreState,
    devTools: true,
    middleware: (getDefaultMiddleware) => {
        const defaultMiddleware = getDefaultMiddleware({
            serializableCheck: {
              ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
            }
        });

        return defaultMiddleware;
    }
});

export default testStore;