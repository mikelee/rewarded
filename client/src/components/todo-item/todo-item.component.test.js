import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import testStore from '../../redux/testStore';
import { Provider } from 'react-redux';
import TodoItem from './todo-item.component';

const props = {
    id: 1,
    text: 'Todo 1',
    completed: true,
    selectedRewardId: null,
    selected: false,
    timestamp: '2023-03-08T06:49:33.913Z'
}

beforeEach(() => {
    render(
        <Provider store={testStore} >
            <TodoItem {...props} />
        </Provider>
    );
})

it('should change TodoItem text', () => {
    const UPDATED_TEXT = 'Updated Text';
    
    const todoText = screen.getByLabelText('todo-text');
    expect(todoText.value).toBe(props.text);
    
    fireEvent.change(todoText, { target: { value: UPDATED_TEXT } });
    expect(todoText.value).toEqual(UPDATED_TEXT);
});