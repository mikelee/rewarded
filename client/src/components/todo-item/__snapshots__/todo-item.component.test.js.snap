// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TodoItem 1`] = `
<div
  className="todo"
  data-testid="todo-1"
  style={
    Object {
      "transform": "translateX(undefinedpx)",
    }
  }
>
  <ToggleButton
    completed={1}
    toggleTodoCompleted={[Function]}
    type="forTodo"
  />
  <form
    className="todo-edit-form"
    id="todo-edit-form-1"
    onBlur={[Function]}
    onSubmit={[Function]}
  >
    <input
      className="todo-edit-form-textfield text-completed"
      defaultValue="first"
      name="text"
      onChange={[Function]}
      placeholder="I want to..."
    />
  </form>
  <form
    className="todo-delete-form"
    onSubmit={[Function]}
  >
    <WithStyles(ForwardRef(IconButton))
      className="todo-icon-button"
      type="submit"
    >
      <ClearIcon
        className="todo-clear-icon"
        fontSize="large"
      />
    </WithStyles(ForwardRef(IconButton))>
  </form>
</div>
`;
